### PR TITLE
Add GLUE finetuning support for multi-sentence classification/regression tasks

### PR DESCRIPTION
**Pull request by RK900**

### CODE REVIEW

File: src/training/cifar_eval.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import argparse
```
Comment (gmittal) [inline src/training/cifar_eval.py]: why is this here?

File: src/training/flava_eval/glue2.py [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+    parser.add_argument(
+        "--pretrained",
+        default='',
+        type=str,
+        help="Use a pretrained CLIP model weights with the specified tag or file path.",
+    )
+    parser.add_argument(
+        "--seed", type=int, default=0, help="Default random seed."
+    )
+    parser.add_argument(
```
Comment (gmittal) [inline src/training/flava_eval/glue2.py:104]: can this be inferred directly from the tokenizer?

File: src/training/flava_eval/glue2.py [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+        "--val-frequency", type=int, default=100, help="How often to run evaluation with val data."
+    )
+    parser.add_argument(
+        "--early-stop-patience", type=int, default=5, help="Early stopping patience."
+    )
+    parser.add_argument(
+        "--early-stop-threshold", type=float, default=0.0, help="Early stopping threshold."
+    )
+    parser.add_argument(
+        "--early-stop-metric-name", type=str, default="accuracy", help="Early stopping metric name."
```
Comment (gmittal) [inline src/training/flava_eval/glue2.py:81]: do you have commands for each task that you used to verify the performance? was accuracy always the early stopping metric?

File: src/training/flava_eval/glue2.py [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+from open_clip.factory import get_tokenizer
+from training.scheduler import cosine_lr
+from training.train import AverageMeter
+
+try:
+    import evaluate
+except ImportError:
+    raise ImportError("Please install HF evaluate: pip install evaluate")
+
+MULTI_PROMPT_TASKS = {"mnli", "mrpc", "qnli", "qqp", "rte", "stsb"}
```
Comment (gmittal) [inline src/training/flava_eval/glue2.py:23]: Do you mean multi-sentence tasks?

File: src/training/flava_eval/glue2.py [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+                'label': label
+            }
+
+
+class TextClassifier(nn.Module):
+    def __init__(self, encoder, embed_dim, num_labels):
+        super().__init__()
+
+        self.encoder = encoder
+        self.logits_proj = nn.Linear(embed_dim, num_labels)
```
Comment (gmittal) [inline src/training/flava_eval/glue2.py:201]: Can we verify that it's just a linear projection head instead of an MLP?

File: src/training/flava_eval/glue2.py [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+
+    def forward(self, text):
+        text_features = self.encoder.encode_text(text)
+        logits = self.logits_proj(text_features)
+        return logits
+
+
+def get_task_metric(task_name):
+    if task_name == "cola":
+        metric = evaluate.load("glue", "stsb")
```
Comment (gmittal) [inline src/training/flava_eval/glue2.py:211]: did this end up being for PCC? if so, can we leave a comment?

File: src/training/flava_eval/glue2.py [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+    samples_seen = 0
+    for batch in dataloader:
+        with torch.no_grad():
+            text = batch["text"].to(device)
+            label = batch["label"].to(device)
+            samples_seen += text.shape[0]
+            logits = model(text)
+            logits = logits.view(-1)
+            label = label.view(-1).float()
+            predictions = torch.sigmoid(logits) > 0.5
```
Comment (gmittal) [inline src/training/flava_eval/glue2.py]: Not all the tasks are binary classification, right? Some are regression and some are multi-class, what do you do here?

File: src/training/flava_eval/glue2.py [hunk 8]
```diff
@@ -0,0 +1,324 @@
...
+        step = epoch * len(data["train"]) + i
+        scheduler(step)
+
+        text = batch["text"].to(device)
+        label = batch["label"].to(device)
+
+        logits = model(text)
+        logits = logits.view(-1)
+        label = label.view(-1).float()
+        loss = nn.functional.binary_cross_entropy_with_logits(logits, label)
```
Comment (gmittal) [inline src/training/flava_eval/glue2.py]: How are you able to use BCE for all of them?

File: src/training/flava_eval/preprocess_glue.py [hunk 9]
```diff
@@ -0,0 +1,37 @@
+from datasets import load_dataset
```
Comment (gmittal) [inline src/training/flava_eval/preprocess_glue.py]: this is no longer necessary right?

File: src/training/vision_eval.py [hunk 10]
```diff
@@ -0,0 +1,67 @@
+import argparse
```
Comment (gmittal) [inline src/training/vision_eval.py]: can also be removed?

File: src/training/flava_eval/glue2.py (hunk 2 above)
Comment (RK900) [reply src/training/flava_eval/glue2.py:104]: maybe push in a later PR?

File: src/training/flava_eval/glue2.py (hunk 3 above)
Comment (RK900) [reply src/training/flava_eval/glue2.py:81]: sts-b uses mcc as early stop

File: src/training/flava_eval/glue2.py (hunk 7 above)
Comment (RK900) [reply src/training/flava_eval/glue2.py]: resolved

File: src/training/flava_eval/glue2.py [hunk 11]
```diff
@@ -0,0 +1,340 @@
...
+
+        if self.patience_counter >= self.patience:
+            return True
+        return False
+
+
+class GLUEDataset(Dataset):
+    """GLUE dataset."""
+
+    def __init__(self, task, split, text_key, label_key, separator_token='<end_of_text>', tokenizer=None):
```
Comment (gmittal) [inline src/training/flava_eval/glue2.py:157]: avoid defaults here, make `separator_token=None`

File: src/training/flava_eval/glue2.py (hunk 5 above)
Comment (RK900) [reply src/training/flava_eval/glue2.py:201]: it is
